### PR TITLE
Preparation for implementing apply of containers desired state

### PR DESCRIPTION
**Pull request by stoyan-zoubev**
[#176] Preparation for implementing apply of desired state for containers update agent

Some preparation work before the real implementation:
- move parse functions from cli main package to common containerm/util package, to be used also by container update agent implementation, updated error messages to be more informative and show the problem
- added additional validation for container - cannot have a privileged container with specified devices

Signed-off-by: Stoyan Zoubev <Stoyan.Zoubev@bosch.io>

### CODE REVIEW

File: containerm/util/containers_validation.go [hunk 1]
```diff
@@ -80,6 +80,9 @@ func ValidateName(name string) error {
 
 // ValidateHostConfig validates the container host configuration
 func ValidateHostConfig(hostConfig *types.HostConfig) error {
+	if hostConfig.Privileged && len(hostConfig.Devices) > 0 {
```
Comment (dimitar-dimitrow) [inline containerm/util/containers_validation.go:83]: Could you please add a test case for this validation check?

File: containerm/util/containers_parse.go [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+	"github.com/eclipse-kanto/container-management/containerm/containers/types"
+	"github.com/eclipse-kanto/container-management/containerm/log"
+)
+
+// ParseDeviceMappings converts string representations of container's device mappings to structured DeviceMapping instances.
+// Format: <host_device>:<container_device>[:propagation_mode].
+// Both path on host and in container must be set.
+// The string representation may contain optional cgroups permissions configuration.
+// Possible cgroup permissions options are “r” (read), “w” (write), “m” (mknod) and all combinations of the three are possible. If not set, “rwm” is default device configuration.
+// Example: /dev/ttyACM0:/dev/ttyUSB0[:rwm].
```
Comment (dimitar-dimitrow) [inline containerm/util/containers_parse.go:29]: Those go docs for the multiple and single container configurations parsing seems very repetitive. Maybe leave the format in the single functions and reference the single function in the multiple function go doc.

**Review by dimitar-dimitrow — CHANGES_REQUESTED** [review]
Unit test should written for the containers_parse.go

File: containerm/util/containers_validation.go (hunk 1 above)
Comment (stoyan-zoubev) [reply containerm/util/containers_validation.go:83]: Validation check added with latest commit.

File: containerm/util/containers_parse_test.go [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+	"github.com/eclipse-kanto/container-management/containerm/pkg/testutil"
+)
+
+type errorTest struct {
+	inputString string
+	errMessage  string
+}
+
+func TestParseDeviceMappings(t *testing.T) {
+	testCases := []struct {
```
[the remark this answers is not in the document]
Comment (stoyan-zoubev) [reply containerm/util/containers_parse_test.go]: I thought the valid input string was self-explainatory and enough. Anyway, test names added as map keys.

**Review by k-gostev — APPROVED** [review]

**Pull request merged**